### PR TITLE
d2format: Expose in CLI under fmt subcommand

### PR DESCRIPTION
**Pull request by nhooyr**
- ci/dev.sh: Add
- lib/xmain: Print usage info in separate log
- d2format: Expose in CLI under fmt subcommand

### CODE REVIEW

**Comment by scsmithr** [comment]
I have a small request on this. Would it be possible to allow providing the input on stdin? I'd like to get formatting integrated into my emacs setup and it's easier to do that with tools that accept file contents on stdin.

**Comment by scsmithr** [comment]
Oh nice, then this works for me.

File: ci/release/changelogs/next.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 #### Features 🚀
 
+- autoformat is supported on the CLI with the `fmt` subcommand.
```
Comment (alixander) [inline ci/release/changelogs/next.md:3]: we should order changelogs by level of importance to user. i think Latex support is higher for example, maybe bump this down a few

File: autofmt.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+)
+
+func autofmt(ctx context.Context, ms *xmain.State) (err error) {
+	defer xdefer.Errorf(&err, "autofmt failed")
+
+	ms.Opts = xmain.NewOpts(ms.Env, ms.Log, ms.Opts.Flags.Args()[1:])
+	if len(ms.Opts.Args) == 0 {
+		return xmain.UsageErrorf("fmt must be passed the file to be formatted")
+	} else if len(ms.Opts.Args) > 1 {
+		return xmain.UsageErrorf("fmt only accepts one argument for the file to be formatted")
```
Comment (alixander) [inline autofmt.go:21]: "accepts only one"

Comment (alixander) [reply autofmt.go:21]: i think u can get rid of the other err msg too and just combine into this. e.g. `if len(ms.Opts.Args) != 1`

File: help.go [hunk 3]
```diff
@@ -29,10 +31,11 @@ Flags:
 
 Subcommands:
   %[1]s layout - Lists available layout engine options with short help
-  %[1]s layout [layout name] - Display long help for a particular layout engine
+  %[1]s layout [name] - Display long help for a particular layout engine
+  %[1]s fmt file.d2 - Autoformat file.d2
```
Comment (alixander) [inline help.go:35]: should we just call it Format? Like, in the IDE plugins, it's auto, but in the CLI the user is just... formatting

**Review by alixander — CHANGES_REQUESTED** [review]

File: ci/release/changelogs/next.md (hunk 1 above)
Comment (nhooyr) [reply ci/release/changelogs/next.md:3]: I very very strongly disagree. autofmt is much more important... latex is a subset of users, autofmt is literally everyone.

File: autofmt.go (hunk 2 above)
Comment (nhooyr) [reply autofmt.go:21]: nah it's not clear if just one, will just update the grammar.

File: help.go (hunk 3 above)
Comment (nhooyr) [reply help.go:35]: true, will change

File: ci/release/changelogs/next.md (hunk 1 above)
Comment (nhooyr) [reply ci/release/changelogs/next.md:3]: We don't have any IDE plugins doing it automatically right now. So people will need to know about it in order to use it. And for plugin authors to implement it.

**Review by alixander — APPROVED** [review]

**Pull request merged**